packages/connect/src/filter.ts

```diff
@@ -118,108 +128,106 @@ export class Filter {
       return false
     }
 
-    // Resolve p2p addresses first
-    if (parsed.address.type === 'p2p') {
-      const p2pAddress = parsed.address
-
-      if (u8aEquals(p2pAddress.node, this.myPublicKey)) {
-        log(`Prevented self-dial using circuit addr. Used addr: ${ma.toString()}`)
-        return false
-      }
-
-      if (u8aEquals(p2pAddress.relayer, this.myPublicKey)) {
-        log(`Prevented dial using self as relay node. Used addr: ${ma.toString()}`)
-        return false
-      }
-
-      return true
-    }
-
-    const address = parsed.address
-
-    if (address.node != undefined && u8aEquals(address.node, this.myPublicKey)) {
-      log(`Prevented self-dial. Used addr: ${ma.toString()}`)
-      return false
-    }
-
-    // If localhost connections are explicitly allowed, do not dial them
-    if (isLocalhost(address.address, address.type) && !this.opts.allowLocalConnections) {
-      // Do not pollute logs by rejecting localhost connections attempts
-      return false
-    }
+    switch (parsed.address.type) {
+      case AddressType.P2P:
+        if (u8aEquals(parsed.address.node, this.myPublicKey)) {
+          log(`Prevented self-dial using circuit addr. Used addr: ${ma.toString()}`)
+          return false
+        }
 
-    // If private address connections are explicitly allowed, do not dial them
-    if (isPrivateAddress(address.address, address.type) && !this.opts.allowPrivateConnections) {
-      // Do not pollute logs by rejecting private address connections attempts
-      return false
-    }
+        if (u8aEquals(parsed.address.relayer, this.myPublicKey)) {
+          log(`Prevented dial using self as relay node. Used addr: ${ma.toString()}`)
+          return false
+        }
 
-    assert(this.announcedAddrs != undefined && this.listeningFamilies != undefined)
```

Comment:
Let's keep this assertion, so that one does need to use (and does not forget to use) `!` later.

Reply:
`assert` is a node.js-only feature. So, for unit testing it is OK, but not for productive code